### DISCUSSION TITLE
Performance issues of Kafka writing to clickhouse

### DISCUSSION THREAD

**Discussion by SoaringCode** [ClickHouse/ClickHouse#67350] [Q&A]
Previously, we used JDBC to write to Clickhouse, but there were performance issues when merging partitions. Now, we want to use Kafka to write to Clickhouse to avoid such problems. Can we solve the problem? The daily data volume is about 20 billion.

**Comment by den-crane** [comment]
It is impossible to answer definitely.
It depends on the cause and the nature of the issues.

**Reply by SoaringCode** [reply]
If using Kafka access, how often is the materialized view written to the target table, and can a threshold be set

**Reply by den-crane** [reply]
* https://github.com/ClickHouse/ClickHouse/pull/11388

https://kb.altinity.com/altinity-kb-integrations/altinity-kb-kafka/altinity-kb-kafka-main-parsing-loop/

**Reply by SoaringCode** [reply]
Thank you very much for your answer, but I still have a question. Regarding the polling data retrieval, it is based on batches. It seems that I have not found the logical rules for retrieving data from each batch. I only found the maximum value batch value, but not the minimum value. Also, how often is it refreshed? My test result shows that writing a data entry will immediately update the target table

**Reply by den-crane** [reply]
stream_flush_interval_ms=7500ms
